cpp/include/cuvs/preprocessing/quantize/binary.hpp

```diff
@@ -31,24 +31,41 @@ namespace cuvs::preprocessing::quantize::binary {
  * @{
  */
 
+/**
+ * @brief quantizer algorithms
+ */
+enum class bit_threshold { zero, mean, sampling_median };
+
+/**
+ * @brief quantizer parameters.
+ */
+struct params {
+  bit_threshold threshold = bit_threshold::mean;
+
+  float sampling_ratio = 0.1;
+};
+
 /**
  * @brief Applies binary quantization transform to given dataset. If a dataset element is positive,
  * set the corresponding bit to 1.
  *
  * Usage example:
  * @code{.cpp}
  * raft::handle_t handle;
+ * cuvs::preprocessing::quantize::binary::params params;
  * auto quantized_dataset = raft::make_device_matrix<uint8_t, int64_t>(handle, samples,
- * features); cuvs::preprocessing::quantize::binary::transform(handle, dataset,
+ * features); cuvs::preprocessing::quantize::binary::transform(handle, params, dataset,
  * quantized_dataset.view());
  * @endcode
  *
  * @param[in] res raft resource
+ * @param[in] params quantization params
  * @param[in] dataset a row-major matrix view on device
  * @param[out] out a row-major matrix view on device
  *
  */
 void transform(raft::resources const& res,
+               const params params,
```

Comment:
This is a breaking change. Please consider keeping still keeping the previous API as a deprecated overload.

Reply:
I kept the previous API.